src/ast/attachpoint_parser.h

```diff
@@ -63,6 +63,6 @@ class AttachPointParser {
 };
 
 // The attachpoints are expanded in their own separate pass.
-Pass CreateParseAttachpointsPass();
+Pass CreateParseAttachpointsPass(bool listing = false);
```

Comment:
Maybe add a warning that `listing` will change the parsing semantics for attach points?